Makefile

```diff
@@ -64,7 +64,9 @@ define NEWLINE
 
 endef
 
-TEST_TIMEOUT ?= 25m
+# 30 minutes is the upper bound defined for all tests, the longer running ones at the time of writing are XDC tests. The
+# 30 minute timeout is also defined in run-tests.yml
+TEST_TIMEOUT ?= 30m
```

Comment:
We actually wanted to leave this timeout lower than the timeout defined in run-tests.yml

If the GHA times out while the tests are still running it will not produce any output logs and we have no way of telling which tests are taking longer than expected.

Reply:
Gave some grace time in https://github.com/temporalio/temporal/pull/6648